packages/daemon/src/mail.js

```diff
@@ -421,29 +454,35 @@ export const makeMailboxMaker = ({
         );
       }
 
-      if (responseName === undefined) {
-        // Behold, recursion:
-        // eslint-disable-next-line no-use-before-define
-        return deliverToRecipient(
-          what,
-          responseName,
-          selfFormulaIdentifier,
-          petStore,
-        );
-      }
-      const responseP = responses.get(responseName);
-      if (responseP !== undefined) {
-        return responseP;
+      if (responseName !== undefined) {
+        const responseP = responses.get(responseName);
+        if (responseP !== undefined) {
+          return responseP;
+        }
       }
+
       // Behold, recursion:
       // eslint-disable-next-line
-      const newResponseP = deliverToRecipient(
+      const recipientResponseP = deliverToRecipient(
+        what,
+        responseName,
+        selfFormulaIdentifier,
+        petStore,
+      );
+      // Send to own inbox.
+      const selfResponseP = respond(
         what,
         responseName,
         selfFormulaIdentifier,
         petStore,
+        recipientFormulaIdentifier,
       );
-      responses.set(responseName, newResponseP);
+      const newResponseP = Promise.race([recipientResponseP, selfResponseP]);
```

Comment:
I’m convinced of this logic. I think we should make a note that in the future, we should probably project the message back to the sender’s inbox with a different form, since the UI and the powers conferred should be different for the sender than the receiver. The sender should be able to cancel the request. Maybe a case can be made that they should be able to preempt the resolution, but I’m not sure.